images/recycler/scripts/recycler.sh

```diff
@@ -22,8 +21,10 @@ if [[ ! -d ${dir} ]]; then
     exit 1
 fi
 
-# shred all files
-find ${dir} -type f -exec shred -fuvz {} \;
+find ${dir} -type f -exec stat -c "%u %g %n"  {} \; |xargs -n 3 bash -c 'sudo -u "#$1" shred  $3' argv0
+
+#delete directories
+find ${dir} -mindepth 1  -type d  -exec stat -c "%n %u %g"  {} \;|sort -k 1 -rg | xargs -n 3 bash -c 'sudo -u "#$2" rm -rf $1' argv0
 
 # remove everything that was left, keeping the directory for re-use as a volume
 if rm -rfv ${dir}/*; then
```

Comment:
What if I created a ${dir}/.ssh directory with my private key? That doesn't seem to get cleaned up....